subsys/bluetooth/services/ots/CMakeLists.txt

```diff
@@ -9,3 +9,8 @@ zephyr_sources_ifdef(
   ots_olcp.c)
 
   zephyr_library_sources_ifdef(CONFIG_BT_OTS_DIR_LIST_OBJ ots_dir_list.c)
+
+zephyr_library_sources_ifdef(
+  CONFIG_BT_OTC
+  otc.c
+  ots_l2cap.c)
```

Comment:
If `CONFIG_BT_OTC` compiles `ots_l2cap.c`, then I think we should guard the individual functionality based on whether `CONFIG_BT_OTC` is enabled or `CONFIG_BT_OTS` is enabled.

Reply:
Within the file, you mean?

Reply:
Yes, e.g. similar to how `gatt.c` has conditional guards for e.g. the gatt client only functions. 

Not sure if it really does a difference in compile size, as I'd assume that the compiler can optimize it though.